### PR TITLE
Add  Style/MethodCallWithArgsParentheses cop

### PR DESCRIPTION
**Pull request by dominh**
This cop checks presence of parentheses in method calls containing parameters.

### Example

```ruby
# bad
array.delete e

# good
array.delete(e)
```

I'm not sure if I should merge it with Style/MethodCallParentheses or something?
Should I make this cop enabled by default?

-----------------

Before submitting the PR make sure the following are checked:

* [ ] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [ ] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [ ] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [ ] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [ ] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

### CODE REVIEW

**Comment by Drenmi** [comment]
This is a great start! 😀 

I opened [an issue](https://github.com/bbatsov/rubocop/issues/3737) for this recently, so you can reference that id in the `CHANGELOG` and commit message.

Also, I think we should not release this cop without a whitelisting configuration option, so that when people (inevitably) start posting issues about some methods, we can point them to that. Also some methods (like `#puts` should probably be in there by default. WDYT? 😀

**Comment by dominh** [comment]
@Drenmi Yes, I fully agree with you. Do we have a default list of Method which needs to be ignored?

**Comment by Drenmi** [comment]
@dominh: Unfortunately not. The ones I can think of, off the top of my head are:

- `#p`, `#puts`, `#print`
- `#require`, `#require_relative` and `#load`
- `#raise` and `#fail`

These all live in `Kernel`, and we should match only receiver `nil`, in case someone implemented these on their own objects.

**Comment by Drenmi** [comment]
There are others that are framework specific, like `render` in Rails, and a bunch of stuff in RSpec, but this is not something RuboCop should know of. Hence the importance for the whitelisting option. 😀

**Comment by bbatsov** [comment]
> I'm not sure if I should merge it with Style/MethodCallParentheses or something?

Probably they should remain separate cops. And we should probably rename the old cop, as it's name is super misleading. 

> Should I make this cop enabled by default?

Sure. If you make it work properly. ;-)

There were a few attempts to make a cop that knew where parens were needed and when it was ok to omit them, but none of them got far enough to be merged and used.

**Comment by bbatsov** [comment]
Any progress here?

**Comment by dominh** [comment]
For now, I'd like to stay with this cop disabled by default and I would to add possibility of ignoring some methods because I haven't any idea what should be ignored by default. In Ruby, we have a lot of DSL, etc. Which looks better when method calls haven't parentheses. I going to look into code tomorrow on the evening. What do you think?

**Comment by dominh** [comment]
I've just update the commit

File: lib/rubocop/cop/style/method_call_with_args_parentheses.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks presence of parentheses in method calls containing
+      # parameters.
+      # As in popular Ruby's frameworks a lot of methods should always be
+      # called without parentheses,
+      # users can ignore them by passing their names to IgnoredMethods option.
```
Comment (bbatsov) [inline lib/rubocop/cop/style/method_call_with_args_parentheses.rb]: Add some code examples here.

File: manual/cops_style.md [hunk 2]
```diff
@@ -2239,7 +2239,37 @@ some_str = 'ala' \
...
+
+Enabled by default | Supports autocorrection
+--- | ---
+Disabled | Yes
+
+This cop checks presence of parentheses in method calls containing parameters.
+As in popular Ruby's frameworks a lot of methods should always be called without parentheses,
+users can ignore them by passing their names to IgnoredMethods option.
+
+### Example
```
Comment (bbatsov) [inline manual/cops_style.md]: Did you edit this file manually? I'm surprised to see examples here when there are none in the cop's description.

**Comment by dominh** [comment]
@bbatsov yeah, it looks like I did these changes manually, strange. Anyway, I've fixed it and updated a commit.

**Comment by bbatsov** [comment]
👍

**Pull request merged**